Devscord.DiscordFramework.UnitTests/Commands/RunnerOfIBotCommandMethodsTests/RunnerOfIBotCommandMethodsTestsService.cs

```diff
@@ -0,0 +1,65 @@
+using Devscord.DiscordFramework.Middlewares.Contexts;
+using Moq;
+using System.Collections.Generic;
+using System;
+using System.Threading.Tasks;
+using System.Text.RegularExpressions;
+using Devscord.DiscordFramework.Framework.Commands;
+using Devscord.DiscordFramework.Framework.Commands.Parsing.Models;
+using System.Reflection;
+
+namespace Devscord.DiscordFramework.UnitTests.Commands.RunnerOfIBotCommandMethodsTests
+{
+    class RunnerOfIBotCommandMethodsTestsService
+    {
+        public static IBotCommandsService GetBotCommandsServiceMock()
+        {
+            var mock = new Mock<IBotCommandsService>();
+            var numberParser = new UniversalNumberParser();
+
+            mock.Setup(m => m.IsDefaultCommand(It.IsAny<BotCommandTemplate>(), It.IsAny<IEnumerable<DiscordRequestArgument>>(), It.IsAny<bool>()))
+                .Returns<BotCommandTemplate, IEnumerable<DiscordRequestArgument>, bool>((template, args, isCommandMatchedWithCustom) => 
+                    new BotCommandsMatchingService(numberParser).IsDefaultCommand(template, args, isCommandMatchedWithCustom));
+
+            mock.Setup(m => m.AreDefaultCommandArgumentsCorrect(It.IsAny<BotCommandTemplate>(), It.IsAny<IEnumerable<DiscordRequestArgument>>()))
+                .Returns<BotCommandTemplate, IEnumerable<DiscordRequestArgument>>((template, args) => 
+                    new BotCommandsMatchingService(numberParser).AreDefaultCommandArgumentsCorrect(template, args));
+
+            mock.Setup(m => m.AreCustomCommandArgumentsCorrect(It.IsAny<BotCommandTemplate>(), It.IsAny<Regex>(), It.IsAny<string>()))
+                .Returns<BotCommandTemplate, Regex, string>((template, customTemplate, input) =>
+                    new BotCommandsMatchingService(numberParser).AreCustomCommandArgumentsCorrect(template, customTemplate, input));
+
+            mock.Setup(m => m.GetCommandTemplate(It.IsAny<Type>()))
+                .Returns<Type>(commandType => new BotCommandsTemplateBuilder().GetCommandTemplate(commandType));
+
+            var botCommandsParsingService = new BotCommandsParsingService(new BotCommandsPropertyConversionService(numberParser), new BotCommandsRequestValueGetterService());
+            mock.Setup(m => m.ParseRequestToCommand(It.IsAny<Type>(), It.IsAny<DiscordRequest>(), It.IsAny<BotCommandTemplate>()))
+                .Returns<Type, DiscordRequest, BotCommandTemplate>((commandType, request, template) => 
+                    botCommandsParsingService.ParseRequestToCommand(commandType, request, template));
+
+            mock.Setup(m => m.ParseCustomTemplate(It.IsAny<Type>(), It.IsAny<BotCommandTemplate>(), It.IsAny<Regex>(), It.IsAny<string>()))
+                .Returns<Type, BotCommandTemplate, Regex, string>((commandType, template, customTemplate, input) => 
+                    botCommandsParsingService.ParseCustomTemplate(commandType, template, customTemplate, input));
+
+            return mock.Object;
```

Comment:
a co jeśli chcielibyśmy podczas testu skonfigurować mocka?

Reply:
Po to umieściłem konfiguracje tego mocka w GetBotCommandsServiceMock, aby nie musieć tego robić w testach. Poza tym po co ci taka możliwość. Spójrz w IBotCommandsService.cs, przecież ten mock jest w całości skonfigurowany, więc gdzie i co chcesz jeszcze konfigurować?